### PR TITLE
Roll pigeon dependency to unblock flutter framework deps roll

### PR DESCRIPTION
**Pull request by aam**
Update pigeon dependency for url_launcher_windows.

This is required to unblock flutter dependency update - see https://github.com/flutter/flutter/pull/130032

### CODE REVIEW

**Comment by aam** [comment]
cc @zanderso

File: packages/url_launcher/url_launcher_windows/CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
-## NEXT
+## 3.0.7
 
+* Update pigeon dependency for url_launcher_windows to "^9.2.4".
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_windows/CHANGELOG.md:3]: Nit: Updates

Per repo style guide.

File: packages/url_launcher/url_launcher_windows/pubspec.yaml [hunk 2]
```diff
@@ -24,5 +24,5 @@ dependencies:
 dev_dependencies:
   flutter_test:
     sdk: flutter
-  pigeon: ^5.0.1
+  pigeon: ^9.2.4
```
Comment (stuartmorgan) [inline packages/url_launcher/url_launcher_windows/pubspec.yaml:27]: Pigeon is a code generator, so updates to it need to update the generated code. Running `update-dependency --pub-package=pigeon --packages/url_launcher_windows` via [the repo tooling](https://github.com/flutter/packages/blob/main/script/tool/README.md) will do the right thing.

(Generally we want to update to the latest Pigeon, but if you really need 9.2.4 for some reason you can add `--version=9.2.4`)

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

File: packages/url_launcher/url_launcher_windows/pubspec.yaml (hunk 2 above)
Comment (aam) [reply packages/url_launcher/url_launcher_windows/pubspec.yaml]: Main reason I chose 9.2.4 was that it is the version that is used by other url_launcher. Going to the latest for this package works too. Of course it is possible to rev up pigeon across all plugins, but I tried to minimize the scope of the change.

File: packages/url_launcher/url_launcher_windows/CHANGELOG.md (hunk 1 above)
Comment (aam) [reply packages/url_launcher/url_launcher_windows/CHANGELOG.md]: Done

File: packages/url_launcher/url_launcher_windows/pubspec.yaml (hunk 2 above)
Comment (stuartmorgan) [reply packages/url_launcher/url_launcher_windows/pubspec.yaml]: > Pigeon is a code generator, so updates to it need to update the generated code. Running `update-dependency --pub-package=pigeon --packages/url_launcher_windows` via [the repo tooling](https://github.com/flutter/packages/blob/main/script/tool/README.md) will do the right thing.

This hasn't been addressed; the generated code update is still not part of the PR.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by aam** [comment]
@stuartmorgan said:
> This hasn't been addressed; the generated code update is still not part of the PR.

Ah, I didn't realize that as I ran the command it reported errors:
```
─➤  PATH=$FH/flutter/bin:$PATH dart run script/tool/bin/flutter_plugin_tools.dart  update-dependency --pub-package=pigeon --packages=url_launcher_windows
Running for url_launcher_windows...
  Updating to "^10.1.2"
  Running pub get...
dart pub get failed (1):
Resolving dependencies...

Because url_launcher_windows requires the Flutter SDK, version solving failed.

Flutter users should run `flutter pub get` instead of `dart pub get`.


Running for url_launcher_windows/example...
  SKIPPING: Does not depend on pigeon


The following packages had errors:
  url_launcher_windows:
    Failed to update pigeon files
See above for full details.
```

**Comment by stuartmorgan** [comment]
Odd, I've never seen that when updating plugin dependencies with the script. I'm not sure why the behavior isn't consistent, but I can fix the call tomorrow.

**Comment by aam** [comment]
> Odd, I've never seen that when updating plugin dependencies with the script. I'm not sure why the behavior isn't consistent, but I can fix the call tomorrow.

I hacked the script so it ran `flutter pub get` and it seems to have worked.

**Review by stuartmorgan — APPROVED** [review]
LGTM

**Comment by aam** [comment]
> LGTM

Is submit-queue failure relevant? It seems to report failure on android tests, which seems to have been skipped.

**Comment by stuartmorgan** [comment]
> Is submit-queue failure relevant? It seems to report failure on android tests, which seems to have been skipped.

`submit-queue` is the tree status, and when you clicked through you were seeing the post-submit status of the last thing that landed in the tree. It's in PRs to prevent people from landing PRs while the tree is red.

Please don't ever land a PR with red checks in a flutter/* unless you have explicit permission from a relevant lead or gardener.

**Pull request merged**